### PR TITLE
ci: fix the `react-native-macos-init` pipeline

### PR DESCRIPTION
**Pull request by tido64**
## Summary:

The pipeline wasn't actually publishing anything and we've been testing against latest on npm all this time.

## Test Plan:

CI should pass.

### CODE REVIEW

File: .ado/jobs/test-react-native-macos-init.yml [hunk 1]
```diff
@@ -83,6 +83,14 @@ jobs:
             npx react-native-macos-init --version latest --overwrite --prerelease
```
Comment (tido64) [inline .ado/jobs/test-react-native-macos-init.yml:83]: @Saadnajmi: This CI only tests latest _published_ version? Shouldn't this be testing `main`?

Comment (Saadnajmi) [reply .ado/jobs/test-react-native-macos-init.yml:83]: I think it publishes to verdacio first?

**Comment by Saadnajmi** [comment]
Probably worth renaming the job to "Verify React Native macOS CLI" or something of that sort since it no longer just tests `react-native-macos-init`?

File: .ado/jobs/test-react-native-macos-init.yml (hunk 1 above)
Comment (Saadnajmi) [reply .ado/jobs/test-react-native-macos-init.yml:83]: Yep, see line 34. It publishes the local version to a verdaccio server and then tests that.

Comment (tido64) [reply .ado/jobs/test-react-native-macos-init.yml:83]: It's clearly not using it:
image

This is what the line should look like: https://github.com/microsoft/react-native-macos/blob/8ea9b6a5a9df035ee7324a2d639e7dd1118a8519/packages/react-native/react-native.config.js#L45

The line number is also wrong.

I can't see what we're using the Verdaccio server at all. Shouldn't there be an `.npmrc` file that points to it somewhere?

Comment (tido64) [reply .ado/jobs/test-react-native-macos-init.yml]: Well, it's no wonder. The publish step doesn't actually do anything. I guess this PR is now about fixing the pipeline…

File: packages/react-native/local-cli/runMacOS/runMacOS.js [hunk 2]
```diff
@@ -162,7 +162,7 @@ function buildProject(sourceDir, xcodeProject, scheme, args) {
   return new Promise((resolve, reject) => {
     const xcodebuildArgs = [
       xcodeProject.isWorkspace ? '-workspace' : '-project',
-      path.join(sourceDir, xcodeProject.path, xcodeProject.name),
+      path.join(sourceDir, xcodeProject.path || '.', xcodeProject.name),
```
Comment (tido64) [inline packages/react-native/local-cli/runMacOS/runMacOS.js:165]: This needs to be backported to 0.74

**Comment by tido64** [comment]
Build currently fails probably due to missing https://github.com/facebook/react-native/commit/774fe0cf6f22f832b756c166599973d9db6a93de

File: packages/react-native/local-cli/runMacOS/runMacOS.js (hunk 2 above)
Comment (tido64) [reply packages/react-native/local-cli/runMacOS/runMacOS.js:165]: - **main:** https://github.com/microsoft/react-native-macos/pull/2237
- **0.74:** https://github.com/microsoft/react-native-macos/pull/2238
- **0.75:** https://github.com/microsoft/react-native-macos/pull/2239

**Comment by Saadnajmi** [comment]
> Build currently fails probably due to missing [facebook@774fe0c](https://github.com/facebook/react-native/commit/774fe0cf6f22f832b756c166599973d9db6a93de)

Spoke offline, I'll merge as is and then rebase #2190 which should fix the pipeline.

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**